extension.js

```diff
@@ -313,9 +317,50 @@ function workarea_for_window(win) {
 }
 
 function target_rect_for_workarea(workarea) {
-    const target_rect = workarea.copy();
-    target_rect.height *= settings.get_double('window-height');
-    return target_rect;
+    const available_width = workarea.width,
+        available_height = workarea.height,
+        full_width = available_width + workarea.x,
+        full_height =  available_height + workarea.y,
+        width = available_width * settings.get_double('window-width'),
+        height = available_height * settings.get_double('window-height'),
+        horizontal_alignment = settings.get_string('window-horizontal-alignment'),
+        vertical_alignment = settings.get_string('window-vertical-alignment');
+
+    // width = Math.min(full_width * settings.get_double('window-width'), available_width),
+    // height = Math.min(full_height * settings.get_double('window-height'), available_height),
```

Comment:
Please remove commented-out code.
And, by the way, `settings.get_double('window-height')` (and, I guess, the same is true for `settings.get_double('window-width')`) can't return values greater than 1 - it's defined in the settings schema.

Reply:
Removed.